src/import/FSHTank.ts

```diff
@@ -364,13 +364,9 @@ export class FSHTank implements Fishable {
           }
           break;
         case Type.Instance:
-          // there are some resource types where "name" is a primitive element.
-          // so, matching on that is also acceptable.
           result = this.getAllInstances().find(
             i =>
-              (i.name === base ||
-                i.id === base ||
-                getNonInstanceValueFromRules(i, 'name', '', 'name') === base) &&
```

Comment:
I had originally added this to support `MasterFisher.fishForMetadata` in the case where it needs to get the `resourceType` for something (in this case, an Instance of ValueSet) fished up from the `FSHTank`:

- In the previous implementation, when doing this, `MasterFisher` would always `fishForMetadata` for an Instance, regardless of what types were originally requested when fishing. This means that it would _not_ use the existing logic in `FSHTank` that is applied when fishing for a ValueSet. This is what caused problems in the regression, since it would find Instances of SearchParameter that had the same name as a resource.
- In the current implementation, `MasterFisher` will `fishForMetadata` using the originally requested types. Since we originally were trying to fish for a ValueSet, `fishForMetadata` will also try to fish for a ValueSet, meaning that we'll be able to get it out of the `FshTank` with its logic for ValueSet, and there's no need to change the logic in `FshTank` for fishing for an Instance.

Which is a whole lot of words to say: the previous fix for this behavior was in `FshTank`, the new fix is in `MasterFisher`.